pontoon/base/models.py

```diff
@@ -1418,6 +1418,16 @@ def check_latest_translation(self, instance):
             instance.latest_translation = self
             instance.save(update_fields=['latest_translation'])
 
+    def discard(self, stats=True):
+        """
+        Discards translation.
+        """
+        self.approved = False
+        self.save()
+
+        TranslatedResource.objects.get(resource=self.entity.resource, locale=self.locale).calculate_stats()
+        self.entity.mark_changed(self.locale)
+
```

Comment:
We should also remove TranslationMemory entry if it doesn't exist anywhere else.